storage/remote/storage.go

```diff
@@ -181,6 +184,7 @@ func (s *Storage) Appender(ctx context.Context) storage.Appender {
 
 // Close the background processing of the storage queues.
 func (s *Storage) Close() error {
+	s.loggerStop()
```

Comment:
we can just call `s.logger.Stop()` here

Reply:
Sorry again for my late response!
In this package, I can't call `s.logger.Stop()` because `s.logger` is `log.Logger`, and `log.Logger` is an interface with only one method `Log()` (code is below).
```go
// .../pkg/mod/github.com/go-kit/log@v0.1.0/log.go:10
type Logger interface {
	Log(keyvals ...interface{}) error
}
```

Reply:
In that case we can just change the field on the storage struct to be `logger *logging.Deduper`

Reply:
Got it. I pushed the change using type `*logging.Deduper` in struct definition and using `s.logger.Stop()`